src/main/java/racingcar/domain/Cars.java

```diff
@@ -0,0 +1,71 @@
+package racingcar.domain;
+
+import static racingcar.message.RacingCarErrorMessage.CAR_LIST_EMPTY;
+import static racingcar.message.RacingCarErrorMessage.DUPLICATED_CAR_NAME;
+import static racingcar.message.RacingCarErrorMessage.NOT_MIN_CAR_COUNT;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import racingcar.util.RacingCarRandomMoveUtils;
+import racingcar.util.RandomUtils;
+
+public class Cars {
+
+    private static final int CAR_COUNT_MINIMUM = 2;
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        isDifferentCarName(cars);
+        minimumCountOfCars(cars.size());
+        this.cars = cars;
+    }
+
+    private void isDifferentCarName(List<Car> cars) {
+        List<String> carName = cars.stream()
+                .map(idx -> idx.getCarName())
+                .collect(Collectors.toList());
+
+        HashSet<String> carNameDuplicateSet = new HashSet<>(carName);
+
+        if (carNameDuplicateSet.size() != carName.size()) {
+            throw new IllegalArgumentException(DUPLICATED_CAR_NAME.getMessage());
+        }
```

Comment:
Stream으로 우아하게 해결할 수 있어요! `distinct()`에 대해 알아보는 건 어떨까요?
```suggestion
        long distinctCount = cars.stream().map(idx -> idx.getCarName()).distinct().count()
```

Reply:
depth이 길이와 가독성 면에서도 훨씬 보기 좋을 것 같네요 !
리뷰 감사드립니다 :)